pycbc/psd/variation.py

```diff
@@ -160,3 +165,220 @@ def find_trigger_value(psd_var, idx, start, sample_rate):
+    srate = int(strain.sample_rate)
+
+    # Fix the step for the PSD estimation and the time to remove at the
+    # edge of the time series.
+    step = 1.0
+    strain_crop = 8.0
+
+    # Find the times of the long segments
+    times_long = numpy.arange(start_time, end_time,
+                              psd_long_segment-2*strain_crop
+                              - segment + step)
+
+    # Set up the empty time series for the PSD variation estimate
+    psd_var = TimeSeries(zeros(int(numpy.floor((end_time - start_time -
+                                                2*strain_crop - segment +
+                                                1) / step))),
+                         delta_t=step, copy=False,
+                         epoch=start_time + strain_crop + segment)
+
+    # Create a bandpass filter between low_freq and high_freq
+    filt = sig.firwin(4*srate, [low_freq, high_freq], pass_zero=False,
+                      window='hann', nyq=srate/2)
+    filt.resize(int(psd_duration*srate))
+    # Fourier transform the filter and take the absolute value to get
+    # rid of the phase. Save the filter as a frequency series.
+    filt = abs(rfft(filt))
+    my_filter = FrequencySeries(filt, delta_f=1./psd_duration,
+                                dtype=fs_dtype)
+
+    ind = 0
+    for tlong in times_long:
+        # Calculate PSD for long segment
+        if tlong + psd_long_segment <= float(end_time):
+            astrain = strain.time_slice(tlong, tlong + psd_long_segment)
+            plong = pycbc.psd.welch(astrain,
+                                    seg_len=int(psd_duration
+                                                * strain.sample_rate),
+                                    seg_stride=int(psd_stride
+                                                   * strain.sample_rate),
+                                    avg_method=psd_avg_method)
+        else:
+            astrain = strain.time_slice(tlong, end_time)
+            plong = pycbc.psd.welch(
+                           strain.time_slice(end_time - psd_long_segment,
+                                             end_time),
+                           seg_len=int(psd_duration * strain.sample_rate),
+                           seg_stride=int(psd_stride * strain.sample_rate),
+                           avg_method=psd_avg_method)
+
+        # Make the weighting filter - bandpass, which weight by f^-7/6,
+        # and whithen. The normalization is chosen so that the variance
+        # will be one if this filter is applied to white noise which
+        # already has a variance of one.
+        freqs = FrequencySeries(plong.sample_frequencies,
+                                delta_f=plong.delta_f,
+                                epoch=plong.epoch, dtype=fs_dtype)
+        fweight = freqs**(-7./6.) * my_filter / numpy.sqrt(plong)
+        fweight[0] = 0.
+        norm = 1. / numpy.sqrt(sum(abs(fweight)**2) / (len(fweight)-1.))
```

Comment:
this expression is hard to read .. maybe split up into 2 lines? eg
`weight_avg = sum(abs(fweight) ** 2.) / (len(fweight) - 1.)` then `norm = 1. / numpy.sqrt(weight_avg)`

also, is `numpy.sqrt()` necessary?  i.e. will it work if you just use `weight_avg ** -0.5` ? to me that would be more readable.

Reply:
i.e. to keep on a single line, `norm = (sum(abs(fweight) ** 2.) / (len(fweight) - 1.)) ** -0.5` if this works OK

Reply:
** -0.5 is definitely better! Thanks for the suggestion!